src/logics/UFLATheory.cc

```diff
@@ -22,7 +27,7 @@ bool UFLATheory::simplify(const vec<PFRef>& formulas, PartitionManager &, int cu
 
 namespace {
     bool isArithmeticSymbol(ArithLogic const & logic, SymRef sym) {
-        return logic.isIF(sym) and not logic.isEquality(sym) and not logic.isBooleanOperator(sym);
+        return logic.isPlus(sym) or logic.isTimes(sym) or logic.isLeq(sym);
```

Comment:
I understand what it is doing here, but what was the logic of checking before? 
Everything is arithmetic symbol if it's IF and not equality or boolean operator?

Reply:
Yes, this was a prototype for the theory combination of UF and LA, but it does not generalize.
Previously, it was saying that every interpreted function that is not equality or boolean connective is an arithmetic symbol. This is obviously not true when we add array-related functions.